app/src/androidTest/java/com/github/sdp_begreen/begreen/matchers/CustomViewAssertionRecyclerView.kt

```diff
@@ -0,0 +1,67 @@
+package com.github.sdp_begreen.begreen.matchers
+
+import android.view.View
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import androidx.test.espresso.NoMatchingViewException
+import androidx.test.espresso.ViewAssertion
+import androidx.test.espresso.matcher.ViewMatchers.assertThat
+import org.hamcrest.Matchers.equalTo
+import org.hamcrest.Matchers.`is`
+import kotlin.test.fail
+
+/**
+ * Custom view assertion, used to assert a [TextView]'s text inside a view contained in a
+ * recycler view at a given position
+ */
+class TextViewAssertionView(
+    private val position: Int,
+    private val id: Int,
+    private val text: String?
+) : ViewAssertion {
+    override fun check(view: View?, noViewFoundException: NoMatchingViewException?) {
+        if (view !is RecyclerView)
+            fail("The view is not a recycler view")
```

Comment:
Yes, that is a very good idea. I actually print the object using `toString()`. Because the `id` is an int. I think we would't really know what it means. What do you think ?